solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1356 @@
+        return uint16(reserveTokens.length);
+    }
+
+    /**
+      * @dev defines a new reserve token for the converter
+      * can only be called by the owner while the converter is inactive
+      *
+      * @param _token   address of the reserve token
+      * @param _weight  reserve weight, represented in ppm, 1-1000000
+    */
+    function addReserve(IERC20Token _token, uint32 _weight)
+        public
+        virtual
+        override
+        ownerOnly
+        inactive
+        validAddress(address(_token))
+        notThis(address(_token))
+        validReserveWeight(_weight)
+    {
+        // validate input
+        require(address(_token) != address(anchor) && reserveIds[_token] == 0, "ERR_INVALID_RESERVE");
+        require(reserveTokenCount() < 2, "ERR_INVALID_RESERVE_COUNT");
+
+        reserveTokens.push(_token);
+        reserveIds[_token] = reserveTokens.length;
+    }
+
+    /**
+      * @dev returns the reserve's weight
+      * added in version 28
+      *
+      * @param _reserveToken    reserve token contract address
+      *
+      * @return reserve weight
+    */
+    function reserveWeight(IERC20Token _reserveToken)
+        public
+        view
+        validReserve(_reserveToken)
+        returns (uint32)
+    {
+        return PPM_RESOLUTION / 2;
+    }
+
+    /**
+      * @dev returns the reserve's balance
+      * note that prior to version 17, you should use 'getConnectorBalance' instead
+      *
+      * @param _reserveToken    reserve token contract address
+      *
+      * @return reserve balance
+    */
+    function reserveBalance(IERC20Token _reserveToken)
+        public
+        override
+        view
+        returns (uint256)
+    {
+        uint256 reserveId = reserveIds[_reserveToken];
+        require(reserveId != 0, "ERR_INVALID_RESERVE");
+        return getReserveBalance(reserveId);        
+    }
+
+    /**
+      * @dev converts a specific amount of source tokens to target tokens
+      * can only be called by the bancor network contract
+      *
+      * @param _sourceToken source ERC20 token
+      * @param _targetToken target ERC20 token
+      * @param _amount      amount of tokens to convert (in units of the source token)
+      * @param _trader      address of the caller who executed the conversion
+      * @param _beneficiary wallet to receive the conversion result
+      *
+      * @return amount of tokens received (in units of the target token)
+    */
+    function convert(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount, address _trader, address payable _beneficiary)
+        public
+        override
+        payable
+        protected
+        only(BANCOR_NETWORK)
+        returns (uint256)
+    {
+        // validate input
+        require(_sourceToken != _targetToken, "ERR_SAME_SOURCE_TARGET");
+
+        // if a whitelist is set, verify that both and trader and the beneficiary are whitelisted
+        require(address(conversionWhitelist) == address(0) ||
```

Comment:
let's get rid of the whitelist functionality in this converter

Reply:
Done